src/main/java/codeu/model/data/User.java

```diff
@@ -105,12 +109,19 @@ public void addToConversation(Conversation conversation) {
     // if it isn't private.
     if (conversation.getIsPrivate()) {
       conversationIdSet.add(conversation.getId());
+      userConversationTitles.add(conversation.getTitle());
     }
   }
 
```

Comment:
Adding a method to recalculate the conversation titles from the conversation ID set would be helpful.

That way, when the user is loaded from the persistent store, you can just call it and update the list of conversation titles.